app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,42 @@
+class FoodsController < ApplicationController
+  load_and_authorize_resource
+  before_action :set_food, only: %i[show destroy]
+
+  def index
+    @foods = current_user.foods
+  end
+
+  def new
+    @food = Food.new
+  end
+
+  def create
+    @food = current_user.foods.new(food_params)
+
+    respond_to do |format|
+      if @food.save
+        format.html { redirect_to food_url(@food), notice: 'Food was successfully created.' }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @food.destroy
+
+    respond_to do |format|
+      format.html { redirect_to foods_url, notice: 'Food was successfully destroyed.' }
+    end
+  end
+
```

Comment:
- [ ] It will be nice to also include a failure path. Just in case food was not successfully destroyed.

Reply:
This change has been implemented well